osquery/tables/system/posix/python_packages.cpp

```diff
@@ -0,0 +1,119 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#include <string>
+#include <fstream>
+
+#include <boost/algorithm/string/trim.hpp>
+
+#include <osquery/core/conversions.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+
+namespace osquery {
+namespace tables {
+
+const int kNumFields = 2;
+const std::set<std::string> kPythonPath = {
+  "/usr/local/lib/python2.7/", "/usr/lib/python2.7/dist-packages",
+};
+
+
+// checks if given set of paths are sub-path of one another
+int checkOverlap() {
+
+  for (const auto& path : kPythonPath) {
+    int count = 0;
+    for (const auto& key : kPythonPath ) {
+      if (count > 1)
+        return -1;
+      else if (key.find(path) != std::string::npos)
+        count += 1;
+    }
+  }
+
+  return 1;
+}
+
+void genPackage(std::string path, Row &r) {
+  std::ifstream fd (path, std::ios::in | std::ios::binary);
```

Comment:
I'd say we probably prefer doing the `osquery::readFile` and `osquery::split` combo, as our `readFile` implementation has some flags that control the "largest possible read" size, so we shouldn't need to worry too much about reading something in that's too terribly large :)

Also doing a `readFile` with a `split` _should_ save you a little of the work you have below, as well as the boost header includes.